analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Monitor.cs

```diff
@@ -0,0 +1,208 @@
+            }
+        }
+
+        public void DoSomethingWithMonitor6()
+        {
+            if (Monitor.TryEnter(obj))// FN
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor7()
+        {
+            if (Monitor.TryEnter(obj)) // Compliant
+            {
+                Monitor.Exit(obj);
+            }
+            else
+            {
+            }
+        }
+
+        public void DoSomethingWithMonitor8(string b)
+        {
+            Monitor.Enter(obj); // Compliant
+            Console.WriteLine(b.Length);
+            Monitor.Exit(obj);
+        }
+
+        public void DoSomethingWithMonitor9(string b)
+        {
+            Monitor.Enter(obj); // Compliant
+            try
+            {
+                Console.WriteLine(b.Length);
+            }
+            finally
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor10(bool b)
+        {
+            bool isAcquired = false;
+            Monitor.Enter(obj, ref isAcquired); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor11()
+        {
+            if (Monitor.TryEnter(obj, 42))// FN
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor12(bool b)
+        {
+            bool isAcquired = false;
+            Monitor.TryEnter(obj, 42, ref isAcquired); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor13()
+        {
+            if (Monitor.TryEnter(obj, new TimeSpan(42)))// FN
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor14(bool b)
+        {
+            bool isAcquired = false;
+            Monitor.TryEnter(obj, new TimeSpan(42), ref isAcquired); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor15()
+        {
+            bool isAcquired = Monitor.TryEnter(obj, 42); // FN
+
+            if (isAcquired)
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor16()
+        {
+            bool isAcquired = Monitor.TryEnter(obj, 42); // Compliant
+
+            if (isAcquired)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor17(bool b)
+        {
+            Monitor.Enter(obj); // FN
+
+            if (b)
+            {
+                throw new Exception();
```

Comment:
Let's add way more `throw` scenarios with various stuff like throw InvalidOperationException unconditionally, catch it, exit lock ONLY there (Compliant)